client/state/data-layer/wpcom-http/pipeline/retry-on-failure/index.jsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import {
+	get,
+	head,
+	sortBy,
+	toPairs,
+} from 'lodash';
+
+const requestQueue = new Map();
+
+export const buildKey = ( { path, apiNamespace, apiVersion, query } ) => JSON.stringify( [
```

Comment:
is this identical to the one in `remove-duplicate-gets`? is it worth sharing the same func and moving it somewhere common